w3af/core/data/context/context/javascript.py

```diff
@@ -36,7 +40,25 @@ class ScriptMultiLineComment(BaseContext):
 
 
 class ScriptStringGeneric(BaseContext):
-    pass
+    ATTR_DELIMITER = None
+
+    def can_break(self):
+        """
+        :return: True if we can break out
+        """
+        escaped = False
+        for c in self.payload:
+            if c == '\\':
+                escaped = not escaped
+            elif c == self.ATTR_DELIMITER and not escaped:
+                return True
+            else:
+                escaped = False
+
+        return False
+
+    def get_payloads(self):
+        return {self.ATTR_DELIMITER, '\\%s' % self.ATTR_DELIMITER}
```

Comment:
This escaping handled only for JavaScript string literals, your example about `HtmlAttrDoubleQuote` that doesn't apply to this case at all.
Hmm, for input `xxx"yyy` and the context `<a href="xxx\"yyy">...` we have a vulnerability since `"` can break `HtmlAttrDoubleQuote`.